### PR TITLE
[[ Bug 18670 ]] Fixed bug preventing all table names being retrieved …

### PR DESCRIPTION
**Pull request by abrisan**
…for MySQL dbs

### CODE REVIEW

**Comment by runrevmark** [comment]
@livecodealex : It occurred to me last night that other db drivers which implement getting tablenames *might* be affected by a similar problem. Could you take a look through them and see?

**Comment by runrevmark** [comment]
@livecodealex The postgres driver suffers an identical bug - could you patch that in the same way in this PR please? :)

File: tests/lcs/core/database/db_getters.livecodescript [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+   end repeat
+end createTables
+
+on TestDBTableGetters
+   TestSkip "skipping DB test","cannot use revOpenDatabase"
+   exit TestDBTableGetters
+   local tArray
+   put generateStrings() into tArray
+   local  tDatabaseID, tNames
+   put revOpenDatabase("mysql","localhost","t2","root","test_password",false,"/tmp/mysql.sock",10,true) into tDatabaseID
```
Comment (runrevmark) [inline tests/lcs/core/database/db_getters.livecodescript]: We should really test all db drivers which have getTables defined here (for when we do get db tests working in the test runner)

**Comment by runrevmark** [comment]
@livecode-vulcan review ok 041f263

**Comment by livecode-vulcan** [comment]
💙 review by @runrevmark ok 041f263dcdf2288e3ce7568d0f4f7837f6d77c35

**Comment by livecode-vulcan** [comment]
😞 test failure 041f263dcdf2288e3ce7568d0f4f7837f6d77c35

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/304)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/310)
* try-community-js-emscripten-sdk1.35: [failure](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/290)
* try-community-universal-ios-iphoneos10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos10.3/builds/233)
* try-community-universal-ios-iphonesimulator10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator10.3/builds/233)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/299)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/304)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/302)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/308)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/302)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/308)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/327)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/298)

**Comment by runrevmark** [comment]
Failure seems to be due to a random internet outage (failed to download a prebuilt when building emscripten).

@livecode-vulcan retry 6ba0f2e

**Comment by runrevmark** [comment]
@livecode-vulcan retry 041f263

**Comment by livecode-vulcan** [comment]
😎 test success 041f263dcdf2288e3ce7568d0f4f7837f6d77c35

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/306)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/312)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/292)
* try-community-universal-ios-iphoneos10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos10.3/builds/235)
* try-community-universal-ios-iphonesimulator10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator10.3/builds/235)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/301)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/306)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/304)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/310)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/304)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/310)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/329)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/300)

**Pull request merged**